{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e47f7fed",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from tqdm import tqdm\n",
    "import os, random\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3a71b25d",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'random' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[3], line 5\u001b[0m\n\u001b[0;32m      1\u001b[0m KMER_LENGTH \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m6\u001b[39m\n\u001b[0;32m      2\u001b[0m MAX_KMER_INPUT_LENGTH \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m512\u001b[39m\n\u001b[1;32m----> 5\u001b[0m \u001b[43mrandom\u001b[49m\u001b[38;5;241m.\u001b[39mseed(\u001b[38;5;241m1000\u001b[39m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'random' is not defined"
     ]
    }
   ],
   "source": [
    "KMER_LENGTH = 6\n",
    "MAX_KMER_INPUT_LENGTH = 512\n",
    "\n",
    "\n",
    "random.seed(1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9649a58a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Accessing the sequence records in all of Amanda's Dataset\n",
    "\n",
    "sequence_records = list()\n",
    "\n",
    "for seq_record in SeqIO.parse(f\"out_s0/2023.03.07_09.24.27_sample_0/reads/read1.fq\", \"fastq\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "\n",
    "    sequence_records.append(seq_record)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "505a3cbc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S0R0/1\n",
      "Seq('CAGTTCCTGCTGATTTACATAAATTGTAAGCTCCATCTCGCTGGGCACATAAAC...TAG')\n",
      "150\n"
     ]
    }
   ],
   "source": [
    "print(sequence_records[0].id)\n",
    "print(repr(sequence_records[0].seq))\n",
    "print(len(sequence_records[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e179ceb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CACAGC ACAGCC CAGCCA AGCCAG GCCAGC CCAGCC CAGC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTAATC TAATCT AATCTA ATCTAG TCTAGT CTAGTA TAGT...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GGAAGA GAAGAG AAGAGG AGAGGG GAGGGA AGGGAA GGGA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGAAAG GAAAGC AAAGCA AAGCAA AGCAAT GCAATC CAAT...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TGACTC GACTCC ACTCCC CTCCCA TCCCAA CCCAAA CCAA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32361</th>\n",
       "      <td>AGGCTG GGCTGA GCTGAG CTGAGG TGAGGC GAGGCA AGGC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32362</th>\n",
       "      <td>ACTGGG CTGGGG TGGGGA GGGGAA GGGAAC GGAACG GAAC...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32363</th>\n",
       "      <td>ACCCTG CCCTGA CCTGAA CTGAAG TGAAGA GAAGAA AAGA...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32364</th>\n",
       "      <td>TTGGCC TGGCCT GGCCTG GCCTGA CCTGAC CTGACC TGAC...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32365</th>\n",
       "      <td>GTGATG TGATGG GATGGT ATGGTG TGGTGT GGTGTT GTGT...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32366 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                sequence  label\n",
       "0      CACAGC ACAGCC CAGCCA AGCCAG GCCAGC CCAGCC CAGC...      0\n",
       "1      CTAATC TAATCT AATCTA ATCTAG TCTAGT CTAGTA TAGT...      1\n",
       "2      GGAAGA GAAGAG AAGAGG AGAGGG GAGGGA AGGGAA GGGA...      1\n",
       "3      CGAAAG GAAAGC AAAGCA AAGCAA AGCAAT GCAATC CAAT...      1\n",
       "4      TGACTC GACTCC ACTCCC CTCCCA TCCCAA CCCAAA CCAA...      1\n",
       "...                                                  ...    ...\n",
       "32361  AGGCTG GGCTGA GCTGAG CTGAGG TGAGGC GAGGCA AGGC...      0\n",
       "32362  ACTGGG CTGGGG TGGGGA GGGGAA GGGAAC GGAACG GAAC...      1\n",
       "32363  ACCCTG CCCTGA CCTGAA CTGAAG TGAAGA GAAGAA AAGA...      1\n",
       "32364  TTGGCC TGGCCT GGCCTG GCCTGA CCTGAC CTGACC TGAC...      1\n",
       "32365  GTGATG TGATGG GATGGT ATGGTG TGGTGT GGTGTT GTGT...      0\n",
       "\n",
       "[32366 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how the fine-tuning data should be formatted\n",
    "pd.read_csv('sampletrain.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "74ea6277",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#anonymous_read_id</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>read_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S0R0/1</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-381809/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S0R0/2</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-381809/2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S0R1/1</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-305183/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S0R1/2</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-305183/2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S0R10/1</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-291411/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666655</th>\n",
       "      <td>S0R99997/2</td>\n",
       "      <td>genome14</td>\n",
       "      <td>526227</td>\n",
       "      <td>NC_014212-38538/2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666656</th>\n",
       "      <td>S0R99998/1</td>\n",
       "      <td>genome17</td>\n",
       "      <td>446468</td>\n",
       "      <td>NC_014211-7028/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666657</th>\n",
       "      <td>S0R99998/2</td>\n",
       "      <td>genome17</td>\n",
       "      <td>446468</td>\n",
       "      <td>NC_014211-7028/2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666658</th>\n",
       "      <td>S0R99999/1</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-222991/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666659</th>\n",
       "      <td>S0R99999/2</td>\n",
       "      <td>genome7</td>\n",
       "      <td>767817</td>\n",
       "      <td>Desgi_Contig329.1-222991/2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>666660 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #anonymous_read_id genome_id  tax_id                     read_id\n",
       "0                  S0R0/1   genome7  767817  Desgi_Contig329.1-381809/1\n",
       "1                  S0R0/2   genome7  767817  Desgi_Contig329.1-381809/2\n",
       "2                  S0R1/1   genome7  767817  Desgi_Contig329.1-305183/1\n",
       "3                  S0R1/2   genome7  767817  Desgi_Contig329.1-305183/2\n",
       "4                 S0R10/1   genome7  767817  Desgi_Contig329.1-291411/1\n",
       "...                   ...       ...     ...                         ...\n",
       "666655         S0R99997/2  genome14  526227           NC_014212-38538/2\n",
       "666656         S0R99998/1  genome17  446468            NC_014211-7028/1\n",
       "666657         S0R99998/2  genome17  446468            NC_014211-7028/2\n",
       "666658         S0R99999/1   genome7  767817  Desgi_Contig329.1-222991/1\n",
       "666659         S0R99999/2   genome7  767817  Desgi_Contig329.1-222991/2\n",
       "\n",
       "[666660 rows x 4 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping = pd.read_csv('out_s0/2023.03.07_09.24.27_sample_0/reads/reads_mapping.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "359fb1ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequences = []\n",
    "labels = []\n",
    "\n",
    "for seq_record in sequence_records:\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
